ownCloud/Client/ClientQueryViewController.swift

```diff
@@ -275,52 +271,13 @@ class ClientQueryViewController: UITableViewController, Themeable {
 				self.navigationController?.pushViewController(ClientQueryViewController(core: self.core, query: OCQuery(forPath: rowItem.path)), animated: true)
 
 			case .file:
-				let itemViewController = DisplayHostViewController(for: rowItem, with: core)
+				let itemViewController = DisplayHostViewController(for: rowItem, with: core, root: query.rootItem!)
 				self.navigationController?.pushViewController(itemViewController, animated: true)
 		}
 
 		tableView.deselectRow(at: indexPath, animated: true)
 	}
 
-	override func tableView(_ tableView: UITableView, trailingSwipeActionsConfigurationForRowAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
```

Comment:
For the re-implementation of row actions, you can use `OCExtensionLocation(ofType: .action, identifier: .tableRow)` as location for building the `ActionContext`.

For actions that shouldn't turn up as row actions, `.tableRow` should not be included in the `ActionSubclass.locations` array, so it doesn't match.